scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -767,11 +770,54 @@
                                        {:job-uuid->reserved-host (apply dissoc job-uuid->reserved-host matched-job-uuids)
                                         :launched-job-uuids (into matched-job-uuids launched-job-uuids)})))
 
+(defn distribute-task-requests-to-compute-clusters
+  "Given a collection of un-matched task requests and a collection of
+  compute clusters, distributes the task requests amongst the compute
+  clusters, using a hash of the pending job's uuid. Returns a
+  compute-cluster->task-request map. That is the API any future
+  improvements need to stick to."
+  [task-requests compute-clusters]
+  (group-by (fn [{:keys [job]}]
+              (nth compute-clusters
+                   (-> job :job/uuid hash (mod (count compute-clusters)))))
+            task-requests))
+
+(defn trigger-autoscaling!
+  "Autoscales the given pool to satisfy the given match failures (i.e. pending jobs), if:
+  - There is at least one failure
+  - There is at least one compute cluster configured to do autoscaling"
+  [failures pool-name compute-clusters]
+  (try
+    (let [task-requests (map #(.. (first %) (getRequest)) failures)
+          num-task-requests (count task-requests)
+          autoscaling-compute-clusters (filter #(cc/autoscaling? % pool-name) compute-clusters)
+          num-autoscaling-compute-clusters (count autoscaling-compute-clusters)]
+      (when (and (pos? num-autoscaling-compute-clusters) (pos? num-task-requests))
+        (let [compute-cluster->task-requests (distribute-task-requests-to-compute-clusters
+                                               task-requests autoscaling-compute-clusters)]
+          (log/info "In" pool-name "pool, autoscaling for" num-task-requests
+                    "un-matched task(s), distributed to compute clusters as follows:"
+                    (->> compute-cluster->task-requests
+                         (pc/map-keys cc/compute-cluster-name)
+                         (pc/map-vals count)))
+          (doseq [[compute-cluster requests-for-cluster] compute-cluster->task-requests]
+            (histograms/update! (histograms/histogram
+                                  (metric-title "autoscale!-tasks"
+                                                pool-name
+                                                compute-cluster))
+                                (count requests-for-cluster))
```

Comment:
Nitpick:
```suggestion
(histograms/update!
  (histograms/histogram (metric-title "autoscale!-tasks" pool-name compute-cluster))
  (count requests-for-cluster))
```

Reply:
Is this just for the formatting? I think the line was too long the way you suggested.

Reply:
Yes, it will be formatted similar to the timer metric call below.